site/src/pages/TemplatePage/TemplateInsightsPage/utils.ts

```diff
@@ -1,50 +1,8 @@
-import {
-  addDays,
-  addHours,
-  format,
-  startOfDay,
-  startOfHour,
-  isToday as isTodayDefault,
-  startOfWeek,
-  endOfDay,
-  endOfWeek,
-  isSunday,
-  subWeeks,
-} from "date-fns";
-
-type GetDateRangeFilterOptions = {
-  startDate: Date;
-  endDate: Date;
-  // Testing purposes
-  now?: Date;
-  isToday?: (date: Date) => boolean;
-};
-
-export function getDateRangeFilter(props: GetDateRangeFilterOptions) {
-  const {
-    startDate,
-    endDate,
-    now = new Date(),
-    isToday = isTodayDefault,
-  } = props;
-
-  return {
-    start_time: toISOLocal(startOfDay(startDate)),
-    end_time: toISOLocal(
-      isToday(endDate)
-        ? startOfHour(addHours(now, 1))
-        : startOfDay(addDays(endDate, 1)),
-    ),
-  };
-}
-
-function toISOLocal(d: Date) {
-  return format(d, "yyyy-MM-dd'T'HH:mm:ssxxx");
-}
+import { startOfDay, subDays } from "date-fns";
 
 export const lastWeeks = (numberOfWeeks: number) => {
   const now = new Date();
-  const startDate = startOfWeek(subWeeks(now, numberOfWeeks));
-  const endDate = isSunday(now) ? endOfDay(now) : endOfWeek(subWeeks(now, 1));
+  const endDate = startOfDay(subDays(now, 1));
```

Comment:
Should this perhaps be just `startOfDay(now)`? Let's say we have `now = 2023-10-10T15:00:00` (Tuesday), `subDays` would put us at `2023-10-09T15:00:00` and `startOfDay` would give us `2023-10-09T00:00:00` meaning we're only looking at data up to and including Sunday.

Reply:
startOfDay does something like 2023-10-09T23:59:59 but the API only accepts dates as "00:00:00" at the end.

Reply:
Ahhh sorry, I missread your comment. I will give it a try